dev/system-test/firestore.ts

```diff
@@ -152,6 +152,25 @@ describe('Firestore class', () => {
         expect(err).to.equal('The client has already been terminated');
       });
   });
+
+  it('throws an error if terminate() is called with active listeners', async () => {
+    const ref = randomCol.doc('doc-1');
+    const unsubscribe = ref.onSnapshot(() => {
+      // No-op
+    });
+
+    await ref.set({});
+    try {
+      await firestore.terminate();
+      return Promise.reject('terminate() should have failed');
```

Comment:
Nit: Would prefer if you just threw an exception here. It's better not to mix Promises and async/await when possible.